From c99f2d947ccb3ad568679d25b6180bfc46906fc7 Mon Sep 17 00:00:00 2001
From: Jamie Lennox <jamielennox@redhat.com>
Date: Fri, 18 Sep 2015 13:17:45 +1000
Subject: Ensure auth_plugin options are in generated CONF

Add the common auth plugin options to the list of options used to
generate sample config files.

Closes-Bug: #1497082
Change-Id: I22d704ebee18b6e9208126812e41aee6cc8331ef
---

--- keystonemiddleware-1.5.2/keystonemiddleware/opts.py.~1~	2015-06-29 08:36:33.000000000 -0700
+++ keystonemiddleware-1.5.2/keystonemiddleware/opts.py	2016-01-18 22:36:05.430322306 -0800
@@ -18,6 +18,8 @@ __all__ = [
 
 import copy
 
+from keystoneclient import auth
+
 import keystonemiddleware.auth_token
 from keystonemiddleware.auth_token import _auth
 from keystonemiddleware.auth_token import _base
@@ -25,7 +27,8 @@ from keystonemiddleware.auth_token impor
 auth_token_opts = [
     (_base.AUTHTOKEN_GROUP,
      keystonemiddleware.auth_token._OPTS +
-     _auth.AuthTokenPlugin.get_options())
+     _auth.AuthTokenPlugin.get_options() +
+     auth.get_common_conf_options())
 ]
 
 
--- keystonemiddleware-1.5.2/keystonemiddleware/tests/test_opts.py.~1~	2015-06-29 08:36:33.000000000 -0700
+++ keystonemiddleware-1.5.2/keystonemiddleware/tests/test_opts.py	2016-01-18 22:36:43.623641221 -0800
@@ -62,7 +62,9 @@ class OptsTestCase(utils.TestCase):
             'include_service_catalog',
             'enforce_token_bind',
             'check_revocations_for_cached',
-            'hash_algorithms'
+            'hash_algorithms',
+            'auth_plugin',
+            'auth_section',
         ]
         opt_names = [o.name for (g, l) in result for o in l]
         self.assertThat(opt_names, matchers.HasLength(len(expected_opt_names)))
